homeassistant/components/light/__init__.py

```diff
@@ -271,6 +273,11 @@ async def async_handle_light_on_service(light, call):
         """
         params = dict(call.data["params"])
 
+        # Zero brightness: Light will be turned off
+        if params.get(ATTR_BRIGHTNESS) == 0:
+            await async_handle_light_off_service(light, call)
+            return
```

Comment:
Moving this will actually break brightness stepping.
So, right now, I can step below zero, and the light will not be turned off.

Reply:
Moved back to original location.

Reply:
https://github.com/home-assistant/core/pull/49376/commits/190402831f40e85f8078560bf1380f79e9486cee